recipe-hooks-project/src/Recipes.js

```diff
@@ -0,0 +1,262 @@
+} from "./components/General";
+import useToggle from "./hooks/useToggleState";
+import {Form, Input} from "./components/Form";
+import {Label} from "./components/Profile";
+import useInputState from "./hooks/useInputState";
+import {INGREDIENTS_RELATIVE_URL} from "./Ingredients";
+import {TAGS_RELATIVE_URL} from "./Tags";
+import {AuthContext} from "./contexts/AuthenticationProvider";
+
+const RECIPES_RELATIVE_URL = "/recipe/recipes/";
+
+const RecipeForm = styled(Form)`
+  width: 40%;
+`;
+const RecipeFormSection = styled.div.attrs(props => ({
+  ...props
+}))`
+  width: 100%;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+`;
+const RecipeFormSelect = styled.select.attrs(props => ({
+  ...props
+}))`
+  width: 50%;
+  margin-top: 20px;
+`;
+const Filters = styled.div.attrs(props => ({
+  ...props
+}))`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  width: 80%;
+`;
+const Filter = styled.div.attrs(props => ({
+  ...props
+}))`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  width: 50%;
+`;
+
+function Recipes() {
+  const {token} = useContext(AuthContext)
+
+  const [recipesList, setRecipes] = useState([]);
+  const [ingredientsList, setIngredients] = useState([]);
+  const [tagsList, setTags] = useState([]);
+  const [selectedIngredients, setSelectedIngredients] = useState([]);
+  const [selectedTags, setSelectedTags] = useState([]);
+  const [isCreatingNewRecipe, toggleCreatingNewRecipe] = useToggle(false);
+  const [title, handleTitleChange] = useInputState("");
+  const [time, handleTimeChange] = useInputState("");
+  const [price, handlePriceChange] = useInputState("");
+  const [link, handleLinkChange] = useInputState("");
+
+  async function fetchRecipes() {
+    console.log("Fetching recipes for: [" + token + "]");
+    let params = {};
+    if (selectedIngredients.length > 0)
+      params = {...params, ingredients: selectedIngredients.join(",")};
+    if (selectedTags.length > 0)
+      params = {...params, tags: selectedTags.join(",")};
+
+    const response = await axios.request({
+      url: RECIPE_API_BASE_URL + RECIPES_RELATIVE_URL,
+      method: "get",
+      headers: {Authorization: "Token " + token},
+      params: {...params}
+    });
+    const recipes = response.data;
+    setRecipes(recipes);
+  }
+
+  useEffect(() => {
+    fetchIngredients();
+    fetchTags();
+    if (!isCreatingNewRecipe)
+      fetchRecipes();
+  }, [isCreatingNewRecipe, selectedIngredients, selectedTags]);
+
+  async function fetchIngredients() {
+    console.log("Fetching ingredients for: [" + token + "]");
+    const response = await axios.request({
+      url: RECIPE_API_BASE_URL + INGREDIENTS_RELATIVE_URL,
+      method: "get",
+      headers: {
+        Authorization: "Token " + token
+      }
+    });
+    const ingredients = response.data;
+    setIngredients(ingredients);
+  }
+
+  async function fetchTags() {
+    console.log("Fetching tags for: [" + token + "]");
```

Comment:
FYI in our web apps it won't be possible to ship code calling `console.log` (I've seen it being used quite a lot in this project)